e2e/config/config.go

```diff
@@ -8,24 +8,23 @@ import (
 )
 
 var (
-	BuildID            = os.Getenv("BUILD_ID")
+	BuildID            = envDefault(os.Getenv("BUILD_ID"), "local")
 	VHDBuildID         = os.Getenv("VHD_BUILD_ID")
-	Subscription       = mustenv("SUBSCRIPTION_ID")
-	Location           = mustenv("LOCATION")
+	SubscriptionID     = envDefault("SUBSCRIPTION_ID", "8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8")
+	Location           = envDefault("LOCATION", "eastus")
 	ResourceGroupName  = "abe2e-" + Location
-	PAT                = os.Getenv("ADO_PAT")
 	ScenariosToRun     = envmap("SCENARIOS_TO_RUN")
 	ScenariosToExclude = envmap("SCENARIOS_TO_EXCLUDE")
 	KeepVMSS           = strings.EqualFold(os.Getenv("KEEP_VMSS"), "true")
-	Azure              = MustNewAzureClient(mustenv("SUBSCRIPTION_ID"))
+	Azure              = MustNewAzureClient(SubscriptionID)
 )
 
-func mustenv(env string) string {
-	result := os.Getenv(env)
-	if result == "" {
-		panic("missing environment variable: " + env)
+func envDefault(env string, defaultValue string) string {
```

Comment:
I'm surprised Go didn't implement this! hehe. 

for the `val == ""` what if the ENV exists but it contains a default "" value which is valid.  

I have seen implementation using https://pkg.go.dev/os#LookupEnv. 

```
func getEnv(key, fallback string) string {
    if value, ok := os.LookupEnv(key); ok {
        return value
    }
    return fallback
}
```

Reply:
`LookupEnv` seems nice :)

Reply:
`LookupEnv ` is important if you want to know that ENV variable is set to an empty string.

In this case, if ENV set to an empty string the fallback won't be used and we will fail something down the line. Is it something you want to achieve?

Reply:
Sorry, I misread the original comment. I don't think we have a single case where `""` is a valid value.
In general, I think it's bad idea to support empty string.

How would like to clarify it? A comment?
The test will fail the someone does it.

Reply:
I would try to find a method name that provides all of the details 
Something like that: 
“getNonEmptyEnv”

Reply:
`envDefault` =>`getEnvWithDefaultIfEmpty`?